habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py

```diff
@@ -505,33 +500,82 @@ def get_all_scenedataset_receptacles(
     return receptacles
 
 
-def import_tri_mesh_ply(ply_file: str) -> TriangleMeshData:
+def import_tri_mesh(mesh_file: str) -> List[mn.trade.MeshData]:
     """
-    Returns a Tuple of (verts,indices) from a ply mesh using magnum trade importer.
+    Returns a list of MeshData objects from a mesh asset using magnum trade importer.
 
-    :param ply_file: The input PLY mesh file. NOTE: must contain only triangles.
+    :param mesh_file: The input meshes file. NOTE: must contain only triangles.
     """
-    manager = mn.trade.ImporterManager()
-    importer = manager.load_and_instantiate("AnySceneImporter")
-    importer.open_file(ply_file)
-
-    # TODO: We don't support mesh merging or multi-mesh parsing currently
-    if importer.mesh_count > 1:
-        raise NotImplementedError(
-            "Importing multi-mesh receptacles (mesh merging or multi-mesh parsing) is not supported."
+
+    _importer.open_file(mesh_file)
+
+    def filter_interleave_mesh(mesh: mn.trade.MeshData) -> mn.trade.MeshData:
```

Comment:
Can this be put outside of `import_tri_mesh` and be made a more general utility? `import_tri_mesh` is already a fairly large method.